[IMP] Calendar: Outlook sync warning (#2925)
odoo/documentation: content/applications/general/calendars/outlook/outlook_calendar.rst

To sync the Outlook Calendar with Odoo's Calendar, a Microsoft Azure account is needed. Creating an
account is free for users who have never tried or paid for Azure. For more information, `click here
<https://azure.microsoft.com/en-us/free/?WT.mc_id=A261C142F>`_.

Refer to `Microsoft's documentation
<https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-create-new-tenant>`_ on
how to set up an Azure AD Tenant (also called an *environment*), which is a representation of an
organization to manage and register apps.

Then, `Register an Application
<https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app>`_,
choosing the appropriate :guilabel:`Supported account type`. Users who wish to connect their
Outlook calendar to Odoo should select the :guilabel:`Accounts in any organizational directory
(Any Azure AD directory - Multitenant and personal Microsoft accounts (e.g. Skype, Xbox)` option
for :guilabel:`Supported account types`.

When configuring the :guilabel:`Redirect URI`, choose :guilabel:`Web` and copy the Odoo database
URI (URL) followed by `/microsoft_account/authentication`. For example,

> `https://www.companyname.odoo.com/microsoft_account/authentication`.

Reviewer: Optional: can also place this in an `.. example::` block

Configuration in Odoo
=====================

In the Odoo database, go to :menuselection:`Settings --> General Settings --> Integrations`
and activate the :guilabel:`Outlook Calendar` setting.

.. image:: outlook_calendar/outlook-calendar-setting.png
   :align: center
   :alt: The "Outlook Calendar" setting activated in Odoo.

From the Microsoft Azure portal, under the :guilabel:`Overview` section of the application, copy
the :guilabel:`Application (Client) ID`, and paste it into the :guilabel:`Client ID` field in Odoo.

.. image:: outlook_calendar/client-id.png
   :align: center
   :alt: The "Client ID" in the Microsoft Azure portal.

In the Microsoft Azure portal, under the :guilabel:`Certificates & secrets` section, copy the
:guilabel:`Client Secret Value` and paste it into the :guilabel:`Client Secret` field in Odoo.

.. image:: outlook_calendar/client-secret-value.png
   :align: center
   :alt: The "Client Secret" token to be copied from Microsoft to Odoo.

Finally, on the Odoo :menuselection:`Settings --> General Settings` page, click :guilabel:`Save`.

.. _outlook_calendar/sync_with_outlook:

Sync with Outlook
=================

> Warnings (before sync)

Reviewer: Optional: some/most/all of this section could be placed in a `.. warning::` block which could call further attention to it, given the yellow background.

======================================
Synchronize Outlook Calendar with Odoo
======================================

Synchronizing a user's Outlook Calendar with Odoo is useful for keeping track of their tasks and
appointments across all related applications.

Register the application with Microsoft Azure
=============================================

To sync the Outlook Calendar with Odoo's Calendar, a Microsoft Azure account is needed. Creating an
account is free for users who have never tried or paid for Azure. For more information, `click here
<https://azure.microsoft.com/en-us/free/?WT.mc_id=A261C142F>`_.

Refer to `Microsoft's documentation <https://docs.microsoft.com/en-us/azure/active-directory/
develop/quickstart-create-new-tenant>`_ on how to set up an Azure AD Tenant (also called an
*environment*), which is a representation of an organization to manage and register apps.

Then, `Register an Application <https://docs.microsoft.com/en-us/azure/active-directory/develop/
quickstart-register-app>`_, choosing the appropriate :guilabel:`Supported account type`. Users who
wish to connect their Outlook calendar to Odoo should select the :guilabel:`Accounts in any

> organizational directory (Any Azure AD directory - Multitenant and personal Microsoft accounts
> (e.g. Skype, Xbox)` option for :guilabel:`Supported account types`.

Reviewer: Missing closing parenthesis.
```suggestion
organizational directory (Any Azure AD directory - Multitenant) and personal Microsoft accounts
(e.g. Skype, Xbox)` option for :guilabel:`Supported account types`.
```

======================================
Synchronize Outlook Calendar with Odoo
======================================

Synchronizing a user's **main** Outlook Calendar with Odoo is useful for keeping track of their
tasks and appointments across all related applications.

Register the application with Microsoft Azure
=============================================

To sync the Outlook Calendar with Odoo's Calendar, a Microsoft Azure account is needed. Creating an
account is free for users who have never tried or paid for Azure. For more information, `click here
<https://azure.microsoft.com/en-us/free/?WT.mc_id=A261C142F>`_.

Refer to `Microsoft's documentation

> <https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-create-new-tenant>`_
> on how to set up an Azure AD Tenant (also called an *environment*), which is a representation of an

Reviewer: Line break slightly sooner.
```suggestion
<https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-create-new-tenant>`_ on
how to set up an Azure AD Tenant (also called an *environment*), which is a representation of an
```

organization to manage and register apps.

Then, `Register an Application
<https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app>`_,
choosing the appropriate supported account type. Users who wish to connect their Outlook calendar
to Odoo should select the :guilabel:`Accounts in any organizational directory (Any Azure AD
directory - Multitenant and personal Microsoft accounts (e.g. Skype, Xbox)` option for
:guilabel:`Supported account types`.

> 

Reviewer: delete extra line break
```suggestion
```

When configuring the platform settings, choose :guilabel:`Web`.

> .. image:: outlook_calendar/platform_configurations.png

Reviewer: Media filenames should be separated with [hypens](https://www.odoo.com/documentation/15.0/contributing/documentation/content_guidelines.html#media-files), not underscores.

   :align: center
   :alt: Adding a platform in Microsoft Azure for Outlook calendar sync in Odoo.

.. important::
   For the :guilabel:`Redirect URI restrictions`, copy the Odoo database URI (URL) followed by
   `/microsoft_account/authentication`. For example, 

>    `(https://www.companyname.odoo.com/microsoft_account/authentication)`.

Reviewer: Not sure parenthesis are needed here with `pre` formatting already in place?
```suggestion
   `https://www.companyname.odoo.com/microsoft_account/authentication`.
```

>    .. image:: outlook_calendar/redirect_URIs.png

Reviewer: Media filenames should be separated with [hypens](https://www.odoo.com/documentation/15.0/contributing/documentation/content_guidelines.html#media-files), not underscores, and not contain capital letters.

      :align: center
      :alt: Odoo's database URI that is accepted when Microsoft returns authentication.

>    For more information on the restrictions and limitations of URIs, `check this page <https://docs.microsoft.com/en-us/azure/active-directory/develop/reply-url>`_.

Reviewer: Need a line break here before 100th char.
```suggestion
   For more information on the restrictions and limitations of URIs, `check this page
   <https://docs.microsoft.com/en-us/azure/active-directory/develop/reply-url>`_.
```

> Regarding credentials, the user **must** :guilabel:`Add a client secret`, which allows Odoo to

Reviewer: Would italicize here for emphasis, as the bold formatting looks like it merges with the guilabel.
```suggestion
Regarding credentials, the user *must* :guilabel:`Add a client secret`, which allows Odoo to
```

authenticate itself, requiring no interaction from the user's side. :guilabel:`Certificates` are
optional.

Since resetting the synchronization can be tricky, Odoo recommends setting the maximum allowed
:guilabel:`expiration date` for the Client Secret (24 months), so there is no need to
re-synchronize soon. 

Configuration in Odoo
=====================

In the Odoo database, go to :menuselection:`Settings --> General Settings` and activate the
:guilabel:`Outlook Calendar` feature.

> .. image:: outlook_calendar/outlook_calendar.png

Reviewer: Media filenames should be separated with [hypens](https://www.odoo.com/documentation/15.0/contributing/documentation/content_guidelines.html#media-files), not underscores.

As well, does this need to be 1200px?

   :align: center
   :alt: Outlook Calendar feature activated in Odoo.

From the Microsoft Azure portal, under the :guilabel:`Overview` section, copy the
:guilabel:`Application (Client) ID`, and paste it into the :guilabel:`Client ID` field in Odoo.

> .. image:: outlook_calendar/application_client_ID.png

Reviewer: Same here, this one's 1200+ pixels and the filename [contains capital letters](https://www.odoo.com/documentation/15.0/contributing/documentation/content_guidelines.html#media-files).

   :align: center
   :alt: Client ID token to be copied from Microsoft to Odoo.

In the Microsoft Azure portal, under the :guilabel:`Certificates & secrets` section, copy the
:guilabel:`Client Secret Value` and paste it into the :guilabel:`Client Secret` field in Odoo.

> .. image:: outlook_calendar/client_secrets.png

Reviewer: [hypens](https://www.odoo.com/documentation/15.0/contributing/documentation/content_guidelines.html#media-files) > underscores in media file name.

   :align: center
   :alt: Client Secret token to be copied from Microsoft to Odoo.

.. _outlook_calendar/sync_with_outlook:

Sync with Outlook
=================

Warnings (before sync)
----------------------

Odoo highly recommends testing the Outlook calendar synchronization on a test database and a test
email address (that is not used for any other purpose) before attempting to sync the desired

> Outlook Calendar with the user’s production database.

Reviewer: This section starting at line 85 and below looks like it got copied in from a doc which uses a serif'd apostrophe instead of the sans-serif version we'd prefer to use in RST.

```suggestion
Outlook Calendar with the user's production database.
```

If the user has any past, present, or future events on their Odoo calendar before syncing their
Outlook calendar, Outlook will treat the events pulled from Odoo's calendar during the sync as new
events, causing an email notification to be sent from Outlook to all the event attendees.

To avoid unwanted emails being sent to all past, present, and future event attendees, the user must
add the events from the Odoo calendar to the Outlook calendar before the first ever sync, delete
the events from Odoo, and then start the sync. 

Even after synchronizing the Odoo Calendar with the Outlook calendar, Outlook will still send a
notification to all event participants every time an event is edited (created, deleted, unarchived,
or event date/time changed), with no exceptions. This is a limitation that cannot be fixed from

> Odoo’s side.

Reviewer: apostrophe
```suggestion
Odoo's side.
```

After one user syncs their Outlook calendar to the Odoo database, unwanted email notifications are

> unavoidable because the first synchronized user’s events will be in the Odoo Calendar. If the Odoo

Reviewer: apostrophe
```suggestion
unavoidable because the first synchronized user's events will be in the Odoo Calendar. If the Odoo
```

database is shared amongst multiple users, and another user wants to sync their Outlook calendar
with Odoo Calendar, Outlook will again pull the existing Odoo Calendar events during the sync and
treat them as new events, causing Outlook to send email invitations to all event attendees.

In summary, once a user synchronizes their Outlook calendar with the Odoo calendar:

- Creating an event in Odoo causes Outlook to send an invitation to all event attendees.
- Deleting an event in Odoo causes Outlook to send a cancellation to all event attendees.
- Unarchiving an event in Odoo causes Outlook to send an invitation to all event attendees.
- Archiving an event in Odoo causes Outlook to send a cancellation to all event attendees.
- Adding a contact to an event causes Outlook to send an invitation to all event attendees.
- Removing a contact from an event causes Outlook to send a cancellation to all event attendees.

If the user decides to synchronize, Odoo suggests that the user sets the maximum allowed expiration

> date for the Client Secret in Microsoft Azure’s portal, so there is no need to re-synchronize soon.

Reviewer: apostrophe
```suggestion
date for the Client Secret in Microsoft Azure's portal, so there is no need to re-synchronize soon.
```

> date for the Client Secret in Microsoft Azure’s portal, so there is no need to re-synchronize soon.

Reviewer: Note: lines 117-121 sounds like a wordier version of what's already written on line 49-51, however some of the language is identical. 

Not sure if that was intentional or written again for emphasis?

Each time the user needs to re-synchronize, the user will have to follow the same process of adding
the events from the Odoo calendar to the Outlook calendar, stopping the sync between Outlook and
Odoo, deleting the events from Odoo, and then starting the sync again.

Sync Odoo Calendar and Outlook
------------------------------

In the Odoo database, go to the :guilabel:`Calendar` module and click :guilabel:`Sync with
Outlook`. The page will redirect to a Microsoft login page, and the user is asked to log in to
their account, if they are not already, and grant the required permissions.

> .. image:: outlook_calendar/sync_with_outlook.png

Reviewer: hyphen > underscore in media filenames.

   :align: center
   :alt: Calendar view and the "Sync with Outlook" button in Odoo Calendar.

The synchronization is a two-way process, meaning that events are reconciled in both accounts
(Outlook and Odoo).

> .. image:: outlook_calendar/synched_with_outlook.png

Reviewer: hyphen > underscore in media filenames.
